tools/frr-reload.py

```diff
@@ -2067,7 +2212,18 @@ def compare_context_objects(newconf, running):
             running.load_from_show_running(args.daemon)
             log.debug("Running Frr Config (Pass #%d)\n%s", x, running.get_lines())
 
-            (lines_to_add, lines_to_del) = compare_context_objects(newconf, running)
+            (lines_to_add, lines_to_del, restart_frr) = compare_context_objects(newconf, running)
+            if restart_frr and is_evpn_enabled():
+                # currently EVPN has heavy dependencies on the BGP default
+                # instance i.e. FRR cannot survive a BGP default instance
+                # delete. so as a workaround we "silently" restart FRR
+                # if the change in config requires a bgp delete (and if
+                # EVPN is enabled).
+                log.info('EVPN is enabled and default instance del needed')
+                log.info('Restarting FRR')
+                subprocess.call('systemctl reset-failed frr.service'.split())
+                subprocess.call('systemctl --no-block restart frr.service'.split())
```

Comment:
We should somehow tell that only systemd is supported?

Reply:
NAK, we support non-systemd.

But it _MIGHT_ be possible to call `service` here, i.e. `service frr restart` - the `service` wrapper is widely(?) available.

(… it AFAIK doesn't have a `--no-block` option though. Not sure if it's worth the effort.)